conference.js

```diff
@@ -2336,6 +2338,102 @@ export default {
         }
     },
 
+    /**
+     * Set a device to be currently used, selected by the user.
+     *
+     * @param {MediaDeviceInfo} device - The device to save.
+     * @returns {boolean} - Returns true in order notifications to be dismissed.
+     * @private
+     */
+    _useDevice(device) {
+        switch (device.kind) {
+        case 'videoinput': {
+            APP.store.dispatch(updateSettings({
+                userSelectedCameraDeviceId: device.deviceId
+            }));
+
+            APP.store.dispatch(setVideoInputDevice(device.deviceId));
+            break;
+        }
+        case 'audioinput': {
+            APP.store.dispatch(updateSettings({
+                userSelectedMicDeviceId: device.deviceId
+            }));
+
+            APP.store.dispatch(setAudioInputDevice(device.deviceId));
+            break;
+        }
+        case 'audiooutput': {
+            setAudioOutputDeviceId(
+                device.deviceId,
+                APP.store.dispatch,
+                true)
+            .then(() => logger.log('changed audio output device'))
+            .catch(err => {
+                logger.warn(
+                    'Failed to change audio output device.',
+                    'Default or previously set audio output device will',
+                    ' be used instead.',
+                    err);
+            });
+            break;
+        }
+        }
+
+        return true;
+    },
+
+    /**
+     * Finds a new device by comparing new and old array of devices and dispatches
+     * notification with the new device.
+     * @param {MediaDeviceInfo[]} newDevices - The array of new devices we received.
+     * @param {MediaDeviceInfo[]} oldDevices - The array of the old devices we have.
+     * @private
+     */
+    _checkAndNotifyForNewDevice(newDevices, oldDevices) {
+        // let's find the first device that exists only in new devices
+        const newDevice = newDevices.find(
```

Comment:
- I think this will find the first element that is new. What happens if there are more than one new devices? Is this possible?

 - Isn't there another solution with better performance? Are you worried about the performance?

 - I'm wondering if you can use `Array.prototype.includes` or `indexOf`. Maybe the code will be more readable?

Reply:
Yep, I see that this is triggered per new device. Ok, I will change it so we handle all new devices in the array ... it may happen different implementation on different browsers ... 

Initially, I tried using both includes and indexOf, but objects are different instances and are not found in that comparison, that's why I did this implementation with two finds so I can compare IDs.
When I plug my camera these are two devices, mic and camera, and I suspect nobody has thousands of devices so we can worry about performance in this case. These are two loops with 5-10 elements.

Reply:
I just pushed a change for that.